specification/source/cxf.rst

```diff
@@ -0,0 +1,290 @@
+properties ``translationSoftware`` and 
+``translationSoftwareVersion`` respectively to the CXF
+block representation. 
+
+Representing Instances in CXF
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Instances of a CDL class shall contain the entire 
+package path of the CDL class, the octothorpe character
+(``#``) followed by the name of the instance in CXF. 
+An instance ("child") of an instance ("parent") shall 
+be referenced by the parent instance’s CXF 
+representation, followed by a period character (``.``)
+and then the child instance’s name. When such a child 
+instance is present in a CDL sequence, the CXF 
+representation of the parent instance shall contain a 
+``hasInstance`` property to the child instance. 
+
+[Example of a CDL instance representation in CXF
+
+CDL:
+  
+.. code-block:: modelica
+
+    within ExamplePackage;
+    block ExampleSeq
+    CDL.Reals.MultiplyByParameter gain(k = 100000 
+        "My gain";
+    end ExampleSeq;
+
+CXF reference to ``gain`` instance: ``ExamplePackage.ExampleSeq#gain``
+CXF reference to ``gain.k`` instance: ``ExamplePackage.ExampleSeq#gain.k``
+CXF property linking ``gain`` and ``gain.k``: ``ExamplePackage.ExampleSeq#gain S231:hasInstance ExamplePackage.ExampleSeq#gain.k .``
+
+Handling Vectors and Expressions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+A CXF translation of a CDL sequence shall optionally 
+include certain configuration options that specify 
+how the translation will handle:
+
+* Vectors: A tool that generates CXF translations 
+  from CDL shall optionally include a configuration
+  indicating whether or not to flatten or 
+  preserve the vector references. By default, 
+  vector references in CDL should be preserved 
+  in CXF. If vector references should be flattened, 
+  an index appearing within square 
+  brackets (``[`` and ``]``) in CDL shall be 
+  appended with the underscore (``_``) character. 
+  For matrices (two dimensions) and arrays with more than
+  two dimensions, each index shall be concatenated
+  with the underscore character (``_``).
+  
+  [For example, ``A[1]`` becomes ``A_1`` and ``B[1 ,2 ,3]``
+  becomes ``B_1_2_3``.
+  ]
+* Expressions: A tool that generates CXF translations 
+  from CDL shall optionally include a configuration 
+  indicating whether or not to evaluate all 
+  expressions in the CDL sequence such as those 
+  within assignment operations, conditional 
+  assignments and arithmetic operations. By default,
+  the expressions shall be preseved in CXF. If the 
+  expressions have to be evaluated and the expressions
+  contain references to a parameter the value of the 
+  parameter will be used in the expression. If the
+  expressions have to be evaluated and expressions 
+  contain references to parameter(s) that does not have 
+  a value binding, then the translation should exit
+  with an error. If there are parameters that do not 
+  have a value binding, the translation process 
+  shall require the user to input the value for
+  such parameters.
```

Comment:
You should delete
> If there are parameters that do not 
  have a value binding, the translation process 
  shall require the user to input the value for
  such parameters.

Everything in this sentence has been said in the previous sentence, i.e., it is an error. What the translation process does about it other than issuing an error need not be written here.